pkg/ccpatch/substitution/ipv6address/substitution_ipv6_address.go

```diff
@@ -0,0 +1,90 @@
+package ccpatch
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/pkg/ccpatch/ipnet"
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/pkg/ccpatch/substitution"
+)
+
+func init() {
+	substitution.RegisterSubstitution(&ipv6Address{})
+}
+
+type ipv6Address struct{}
+
+var _ substitution.Handler = &ipv6Address{}
+
+func (i *ipv6Address) Type() string {
+	return "ipv6Address"
+}
+
+func (i *ipv6Address) WriteState(identifier substitution.Identifier, gs *substitution.GlobalState, sub substitution.Substitution) error {
+	value, ok := sub.AdditionalProperties["cidr"]
+	if !ok {
+		return substitution.ErrMissingCIDR
+	}
+
+	if gs.Exists(identifier, sub.Key) {
+		return nil
+	}
+
+	used := []string{}
```

Comment:
highlighted, regarding the prealloc note.

Reply:
See: https://github.com/ionos-cloud/crossplane-provider-ionoscloud/pull/235#discussion_r1625726947

Reply:
same as [above](https://github.com/ionos-cloud/crossplane-provider-ionoscloud/pull/235#discussion_r1627386657)